litellm/llms/vertex_ai.py

```diff
@@ -261,6 +261,7 @@ def completion(
     litellm_params=None,
     logger_fn=None,
     acompletion: bool = False,
+    credentials = None,
```

Comment:
you don't need to explicitly add this - this would then require an update in main.py too. 

If you pass in credentials, it should be accessible inside vertex_ai as part of optional_params

Reply:
something like `optional_params["credentials"]` would exist if you pass it in as a param

Reply:
@krrishdholakia This isn't working, code example below and what I think is going on:

**`vertex_ai.py`**

```python
vertexai.init(
    project=vertex_project, location=vertex_location, credentials=creds
)
```

Vertex AI has to be initialized like so and creds is currently only set via a single `creds` value. Before this PR that's only done via on-disk application credentials. 

Here's some failing demo code using your proposed solution (which is much more succinct and elegant, IMO).

**failing code**

```python
import base64
import json
import litellm
import os
from google.oauth2 import service_account

litellm.vertex_project = "blah-blah"  # This is my project in the test code
litellm.vertex_location = "us-central1"  
# ignore - this simply grabs the service account that is stored as a base64 string
service_account_json_string_b64: str = (
    os.getenv("GCP_APPLICATION_DEFAULT_CREDENTIALS") or ""
)

service_account_json_string = base64.b64decode(
    service_account_json_string_b64
).decode("utf-8")

# relevant code below
service_account_json = json.loads(service_account_json_string)

credentials = service_account.Credentials.from_service_account_info(
    service_account_json
)

response = litellm.completion(
    model="gemini-pro",
    messages=[
        {
            "role": "user",
            "content": "Tell me a joke about Douglas Adams in 42 characters.",
        }
    ],
    optional_params={"credentials": credentials},
)

print(response.choices[0].message.content)
```
I'll paste the debug output and stacktrace in the proceeding comment.

For this (`optional_params["credentials"]`) to correctly work we could check the value of the `credentials` key in `optional_params` inside the `completion` function for the `vertex_ai` module (and remove the previous code I modified), where VertexAI is initialized. 

If this isn't what you're looking for I'm happy to help implement the functionality how you envision it. I anticipate leveraging LiteLLM heavily in my businesses so more than happy to get my hands dirty here. 

```python
creds, _ = google.auth.default(quota_project_id=vertex_project) if credentials is None else (credentials, None)
```